overlord/hookstate/hooks.go

```diff
@@ -375,4 +368,5 @@ func setupHooks(hookMgr *HookManager) {
 	hookMgr.Register(regexp.MustCompile("^pre-refresh$"), handlerGenerator)
 	hookMgr.Register(regexp.MustCompile("^remove$"), handlerGenerator)
 	hookMgr.Register(regexp.MustCompile("^gate-auto-refresh$"), gateAutoRefreshHandlerGenerator)
+	hookMgr.Register(regexp.MustCompile("^.+-view-changed$"), ViewChangedHandlerGenerator)
```

Comment:
I think I asked this in the original PR, why can't we do this registration in registrystate itself?

Reply:
it would cause an import cycle. `registrystate` needs to know about `hookstate.Context` to be able to tell what to do